File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-cd-5-4/IndexReferenceInformation.cpp
// IndexReferenceInformation.cpp 
#include "IndexReferenceInformation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

IndexReferenceInformation::IndexReferenceInformation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //indexNameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexNameNode = xmlNode->FirstChildElement("indexName");

   if(indexNameNode){indexNameIsNull_ = false;}
   else{indexNameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexNameNode , address : " << indexNameNode << std::endl;
   #endif
   if(indexNameNode)
   {
      if(indexNameNode->Attribute("href") || indexNameNode->Attribute("id"))
      {
          if(indexNameNode->Attribute("id"))
          {
             indexNameIDRef_ = indexNameNode->Attribute("id");
             indexName_ = boost::shared_ptr<IndexName>(new IndexName(indexNameNode));
             indexName_->setID(indexNameIDRef_);
             IDManager::instance().setID(indexNameIDRef_,indexName_);
          }
          else if(indexNameNode->Attribute("href")) { indexNameIDRef_ = indexNameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexName_ = boost::shared_ptr<IndexName>(new IndexName(indexNameNode));}
   }

   //indexIdNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexIdNode = xmlNode->FirstChildElement("indexId");

   if(indexIdNode){indexIdIsNull_ = false;}
   else{indexIdIsNull_ = true;}

   if(indexIdNode)
   {
      for(indexIdNode; indexIdNode; indexIdNode = indexIdNode->NextSiblingElement("indexId")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexIdNode , address : " << indexIdNode << std::endl;
          #endif
          if(indexIdNode->Attribute("href") || indexIdNode->Attribute("id"))
          {
              if(indexIdNode->Attribute("id"))
              {
                  indexIdIDRef_ = indexIdNode->Attribute("id");
                  indexId_.push_back(boost::shared_ptr<IndexId>(new IndexId(indexIdNode)));
                  indexId_.back()->setID(indexIdIDRef_);
                  IDManager::instance().setID(indexIdIDRef_, indexId_.back());
              }
              else if(indexIdNode->Attribute("href")) { indexIdIDRef_ = indexIdNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { indexId_.push_back(boost::shared_ptr<IndexId>(new IndexId(indexIdNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexIdNode , address : " << indexIdNode << std::endl;
       #endif
   }

   //indexSeriesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexSeriesNode = xmlNode->FirstChildElement("indexSeries");

   if(indexSeriesNode){indexSeriesIsNull_ = false;}
   else{indexSeriesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexSeriesNode , address : " << indexSeriesNode << std::endl;
   #endif
   if(indexSeriesNode)
   {
      if(indexSeriesNode->Attribute("href") || indexSeriesNode->Attribute("id"))
      {
          if(indexSeriesNode->Attribute("id"))
          {
             indexSeriesIDRef_ = indexSeriesNode->Attribute("id");
             indexSeries_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(indexSeriesNode));
             indexSeries_->setID(indexSeriesIDRef_);
             IDManager::instance().setID(indexSeriesIDRef_,indexSeries_);
          }
          else if(indexSeriesNode->Attribute("href")) { indexSeriesIDRef_ = indexSeriesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexSeries_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(indexSeriesNode));}
   }

   //indexAnnexVersionNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexAnnexVersionNode = xmlNode->FirstChildElement("indexAnnexVersion");

   if(indexAnnexVersionNode){indexAnnexVersionIsNull_ = false;}
   else{indexAnnexVersionIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexAnnexVersionNode , address : " << indexAnnexVersionNode << std::endl;
   #endif
   if(indexAnnexVersionNode)
   {
      if(indexAnnexVersionNode->Attribute("href") || indexAnnexVersionNode->Attribute("id"))
      {
          if(indexAnnexVersionNode->Attribute("id"))
          {
             indexAnnexVersionIDRef_ = indexAnnexVersionNode->Attribute("id");
             indexAnnexVersion_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(indexAnnexVersionNode));
             indexAnnexVersion_->setID(indexAnnexVersionIDRef_);
             IDManager::instance().setID(indexAnnexVersionIDRef_,indexAnnexVersion_);
          }
          else if(indexAnnexVersionNode->Attribute("href")) { indexAnnexVersionIDRef_ = indexAnnexVersionNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexAnnexVersion_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(indexAnnexVersionNode));}
   }

   //indexAnnexDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexAnnexDateNode = xmlNode->FirstChildElement("indexAnnexDate");

   if(indexAnnexDateNode){indexAnnexDateIsNull_ = false;}
   else{indexAnnexDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexAnnexDateNode , address : " << indexAnnexDateNode << std::endl;
   #endif
   if(indexAnnexDateNode)
   {
      if(indexAnnexDateNode->Attribute("href") || indexAnnexDateNode->Attribute("id"))
      {
          if(indexAnnexDateNode->Attribute("id"))
          {
             indexAnnexDateIDRef_ = indexAnnexDateNode->Attribute("id");
             indexAnnexDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(indexAnnexDateNode));
             indexAnnexDate_->setID(indexAnnexDateIDRef_);
             IDManager::instance().setID(indexAnnexDateIDRef_,indexAnnexDate_);
          }
          else if(indexAnnexDateNode->Attribute("href")) { indexAnnexDateIDRef_ = indexAnnexDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexAnnexDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(indexAnnexDateNode));}
   }

   //indexAnnexSourceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexAnnexSourceNode = xmlNode->FirstChildElement("indexAnnexSource");

   if(indexAnnexSourceNode){indexAnnexSourceIsNull_ = false;}
   else{indexAnnexSourceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexAnnexSourceNode , address : " << indexAnnexSourceNode << std::endl;
   #endif
   if(indexAnnexSourceNode)
   {
      if(indexAnnexSourceNode->Attribute("href") || indexAnnexSourceNode->Attribute("id"))
      {
          if(indexAnnexSourceNode->Attribute("id"))
          {
             indexAnnexSourceIDRef_ = indexAnnexSourceNode->Attribute("id");
             indexAnnexSource_ = boost::shared_ptr<IndexAnnexSource>(new IndexAnnexSource(indexAnnexSourceNode));
             indexAnnexSource_->setID(indexAnnexSourceIDRef_);
             IDManager::instance().setID(indexAnnexSourceIDRef_,indexAnnexSource_);
          }
          else if(indexAnnexSourceNode->Attribute("href")) { indexAnnexSourceIDRef_ = indexAnnexSourceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexAnnexSource_ = boost::shared_ptr<IndexAnnexSource>(new IndexAnnexSource(indexAnnexSourceNode));}
   }

   //excludedReferenceEntityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* excludedReferenceEntityNode = xmlNode->FirstChildElement("excludedReferenceEntity");

   if(excludedReferenceEntityNode){excludedReferenceEntityIsNull_ = false;}
   else{excludedReferenceEntityIsNull_ = true;}

   if(excludedReferenceEntityNode)
   {
      for(excludedReferenceEntityNode; excludedReferenceEntityNode; excludedReferenceEntityNode = excludedReferenceEntityNode->NextSiblingElement("excludedReferenceEntity")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excludedReferenceEntityNode , address : " << excludedReferenceEntityNode << std::endl;
          #endif
          if(excludedReferenceEntityNode->Attribute("href") || excludedReferenceEntityNode->Attribute("id"))
          {
              if(excludedReferenceEntityNode->Attribute("id"))
              {
                  excludedReferenceEntityIDRef_ = excludedReferenceEntityNode->Attribute("id");
                  excludedReferenceEntity_.push_back(boost::shared_ptr<LegalEntity>(new LegalEntity(excludedReferenceEntityNode)));
                  excludedReferenceEntity_.back()->setID(excludedReferenceEntityIDRef_);
                  IDManager::instance().setID(excludedReferenceEntityIDRef_, excludedReferenceEntity_.back());
              }
              else if(excludedReferenceEntityNode->Attribute("href")) { excludedReferenceEntityIDRef_ = excludedReferenceEntityNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { excludedReferenceEntity_.push_back(boost::shared_ptr<LegalEntity>(new LegalEntity(excludedReferenceEntityNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- excludedReferenceEntityNode , address : " << excludedReferenceEntityNode << std::endl;
       #endif
   }

   //trancheNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* trancheNode = xmlNode->FirstChildElement("tranche");

   if(trancheNode){trancheIsNull_ = false;}
   else{trancheIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- trancheNode , address : " << trancheNode << std::endl;
   #endif
   if(trancheNode)
   {
      if(trancheNode->Attribute("href") || trancheNode->Attribute("id"))
      {
          if(trancheNode->Attribute("id"))
          {
             trancheIDRef_ = trancheNode->Attribute("id");
             tranche_ = boost::shared_ptr<Tranche>(new Tranche(trancheNode));
             tranche_->setID(trancheIDRef_);
             IDManager::instance().setID(trancheIDRef_,tranche_);
          }
          else if(trancheNode->Attribute("href")) { trancheIDRef_ = trancheNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { tranche_ = boost::shared_ptr<Tranche>(new Tranche(trancheNode));}
   }

   //settledEntityMatrixNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* settledEntityMatrixNode = xmlNode->FirstChildElement("settledEntityMatrix");

   if(settledEntityMatrixNode){settledEntityMatrixIsNull_ = false;}
   else{settledEntityMatrixIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settledEntityMatrixNode , address : " << settledEntityMatrixNode << std::endl;
   #endif
   if(settledEntityMatrixNode)
   {
      if(settledEntityMatrixNode->Attribute("href") || settledEntityMatrixNode->Attribute("id"))
      {
          if(settledEntityMatrixNode->Attribute("id"))
          {
             settledEntityMatrixIDRef_ = settledEntityMatrixNode->Attribute("id");
             settledEntityMatrix_ = boost::shared_ptr<SettledEntityMatrix>(new SettledEntityMatrix(settledEntityMatrixNode));
             settledEntityMatrix_->setID(settledEntityMatrixIDRef_);
             IDManager::instance().setID(settledEntityMatrixIDRef_,settledEntityMatrix_);
          }
          else if(settledEntityMatrixNode->Attribute("href")) { settledEntityMatrixIDRef_ = settledEntityMatrixNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { settledEntityMatrix_ = boost::shared_ptr<SettledEntityMatrix>(new SettledEntityMatrix(settledEntityMatrixNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<IndexName> IndexReferenceInformation::getIndexName()
{
   if(!this->indexNameIsNull_){
        if(indexNameIDRef_ != ""){
             return boost::shared_static_cast<IndexName>(IDManager::instance().getID(indexNameIDRef_));
        }else{
             return this->indexName_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<IndexName>();
   }
}
std::vector<boost::shared_ptr<IndexId>> IndexReferenceInformation::getIndexId()
{
   if(!this->indexIdIsNull_){
        if(indexIdIDRef_ != ""){
             return this->indexId_;
        }else{
             return this->indexId_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<IndexId>>();
   }
}
boost::shared_ptr<XsdTypePositiveInteger> IndexReferenceInformation::getIndexSeries()
{
   if(!this->indexSeriesIsNull_){
        if(indexSeriesIDRef_ != ""){
             return boost::shared_static_cast<XsdTypePositiveInteger>(IDManager::instance().getID(indexSeriesIDRef_));
        }else{
             return this->indexSeries_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypePositiveInteger>();
   }
}
boost::shared_ptr<XsdTypePositiveInteger> IndexReferenceInformation::getIndexAnnexVersion()
{
   if(!this->indexAnnexVersionIsNull_){
        if(indexAnnexVersionIDRef_ != ""){
             return boost::shared_static_cast<XsdTypePositiveInteger>(IDManager::instance().getID(indexAnnexVersionIDRef_));
        }else{
             return this->indexAnnexVersion_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypePositiveInteger>();
   }
}
boost::shared_ptr<XsdTypeDate> IndexReferenceInformation::getIndexAnnexDate()
{
   if(!this->indexAnnexDateIsNull_){
        if(indexAnnexDateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDate>(IDManager::instance().getID(indexAnnexDateIDRef_));
        }else{
             return this->indexAnnexDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDate>();
   }
}
boost::shared_ptr<IndexAnnexSource> IndexReferenceInformation::getIndexAnnexSource()
{
   if(!this->indexAnnexSourceIsNull_){
        if(indexAnnexSourceIDRef_ != ""){
             return boost::shared_static_cast<IndexAnnexSource>(IDManager::instance().getID(indexAnnexSourceIDRef_));
        }else{
             return this->indexAnnexSource_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<IndexAnnexSource>();
   }
}
std::vector<boost::shared_ptr<LegalEntity>> IndexReferenceInformation::getExcludedReferenceEntity()
{
   if(!this->excludedReferenceEntityIsNull_){
        if(excludedReferenceEntityIDRef_ != ""){
             return this->excludedReferenceEntity_;
        }else{
             return this->excludedReferenceEntity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<LegalEntity>>();
   }
}
boost::shared_ptr<Tranche> IndexReferenceInformation::getTranche()
{
   if(!this->trancheIsNull_){
        if(trancheIDRef_ != ""){
             return boost::shared_static_cast<Tranche>(IDManager::instance().getID(trancheIDRef_));
        }else{
             return this->tranche_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Tranche>();
   }
}
boost::shared_ptr<SettledEntityMatrix> IndexReferenceInformation::getSettledEntityMatrix()
{
   if(!this->settledEntityMatrixIsNull_){
        if(settledEntityMatrixIDRef_ != ""){
             return boost::shared_static_cast<SettledEntityMatrix>(IDManager::instance().getID(settledEntityMatrixIDRef_));
        }else{
             return this->settledEntityMatrix_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<SettledEntityMatrix>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-cd-5-4/IndexReferenceInformation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class IndexReferenceInformation : ISerialized
    {
        public IndexReferenceInformation(XmlNode xmlNode)
        {
            XmlNode indexNameNode = xmlNode.SelectSingleNode("indexName");
            
            if (indexNameNode != null)
            {
                if (indexNameNode.Attributes["href"] != null || indexNameNode.Attributes["id"] != null) 
                {
                    if (indexNameNode.Attributes["id"] != null) 
                    {
                        indexNameIDRef_ = indexNameNode.Attributes["id"].Value;
                        IndexName ob = new IndexName(indexNameNode);
                        IDManager.SetID(indexNameIDRef_, ob);
                    }
                    else if (indexNameNode.Attributes["href"] != null)
                    {
                        indexNameIDRef_ = indexNameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexName_ = new IndexName(indexNameNode);
                    }
                }
                else
                {
                    indexName_ = new IndexName(indexNameNode);
                }
            }
            
        
            XmlNodeList indexIdNodeList = xmlNode.SelectNodes("indexId");
            
            if (indexIdNodeList != null)
            {
                this.indexId_ = new List<IndexId>();
                foreach (XmlNode item in indexIdNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            indexIdIDRef_ = item.Attributes["id"].Value;
                            indexId_.Add(new IndexId(item));
                            IDManager.SetID(indexIdIDRef_, indexId_[indexId_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            indexIdIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        indexId_.Add(new IndexId(item));
                        }
                    }
                    else
                    {
                        indexId_.Add(new IndexId(item));
                    }
                }
            }
            
        
            XmlNode indexSeriesNode = xmlNode.SelectSingleNode("indexSeries");
            
            if (indexSeriesNode != null)
            {
                if (indexSeriesNode.Attributes["href"] != null || indexSeriesNode.Attributes["id"] != null) 
                {
                    if (indexSeriesNode.Attributes["id"] != null) 
                    {
                        indexSeriesIDRef_ = indexSeriesNode.Attributes["id"].Value;
                        XsdTypePositiveInteger ob = new XsdTypePositiveInteger(indexSeriesNode);
                        IDManager.SetID(indexSeriesIDRef_, ob);
                    }
                    else if (indexSeriesNode.Attributes["href"] != null)
                    {
                        indexSeriesIDRef_ = indexSeriesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexSeries_ = new XsdTypePositiveInteger(indexSeriesNode);
                    }
                }
                else
                {
                    indexSeries_ = new XsdTypePositiveInteger(indexSeriesNode);
                }
            }
            
        
            XmlNode indexAnnexVersionNode = xmlNode.SelectSingleNode("indexAnnexVersion");
            
            if (indexAnnexVersionNode != null)
            {
                if (indexAnnexVersionNode.Attributes["href"] != null || indexAnnexVersionNode.Attributes["id"] != null) 
                {
                    if (indexAnnexVersionNode.Attributes["id"] != null) 
                    {
                        indexAnnexVersionIDRef_ = indexAnnexVersionNode.Attributes["id"].Value;
                        XsdTypePositiveInteger ob = new XsdTypePositiveInteger(indexAnnexVersionNode);
                        IDManager.SetID(indexAnnexVersionIDRef_, ob);
                    }
                    else if (indexAnnexVersionNode.Attributes["href"] != null)
                    {
                        indexAnnexVersionIDRef_ = indexAnnexVersionNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexAnnexVersion_ = new XsdTypePositiveInteger(indexAnnexVersionNode);
                    }
                }
                else
                {
                    indexAnnexVersion_ = new XsdTypePositiveInteger(indexAnnexVersionNode);
                }
            }
            
        
            XmlNode indexAnnexDateNode = xmlNode.SelectSingleNode("indexAnnexDate");
            
            if (indexAnnexDateNode != null)
            {
                if (indexAnnexDateNode.Attributes["href"] != null || indexAnnexDateNode.Attributes["id"] != null) 
                {
                    if (indexAnnexDateNode.Attributes["id"] != null) 
                    {
                        indexAnnexDateIDRef_ = indexAnnexDateNode.Attributes["id"].Value;
                        XsdTypeDate ob = new XsdTypeDate(indexAnnexDateNode);
                        IDManager.SetID(indexAnnexDateIDRef_, ob);
                    }
                    else if (indexAnnexDateNode.Attributes["href"] != null)
                    {
                        indexAnnexDateIDRef_ = indexAnnexDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexAnnexDate_ = new XsdTypeDate(indexAnnexDateNode);
                    }
                }
                else
                {
                    indexAnnexDate_ = new XsdTypeDate(indexAnnexDateNode);
                }
            }
            
        
            XmlNode indexAnnexSourceNode = xmlNode.SelectSingleNode("indexAnnexSource");
            
            if (indexAnnexSourceNode != null)
            {
                if (indexAnnexSourceNode.Attributes["href"] != null || indexAnnexSourceNode.Attributes["id"] != null) 
                {
                    if (indexAnnexSourceNode.Attributes["id"] != null) 
                    {
                        indexAnnexSourceIDRef_ = indexAnnexSourceNode.Attributes["id"].Value;
                        IndexAnnexSource ob = new IndexAnnexSource(indexAnnexSourceNode);
                        IDManager.SetID(indexAnnexSourceIDRef_, ob);
                    }
                    else if (indexAnnexSourceNode.Attributes["href"] != null)
                    {
                        indexAnnexSourceIDRef_ = indexAnnexSourceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexAnnexSource_ = new IndexAnnexSource(indexAnnexSourceNode);
                    }
                }
                else
                {
                    indexAnnexSource_ = new IndexAnnexSource(indexAnnexSourceNode);
                }
            }
            
        
            XmlNodeList excludedReferenceEntityNodeList = xmlNode.SelectNodes("excludedReferenceEntity");
            
            if (excludedReferenceEntityNodeList != null)
            {
                this.excludedReferenceEntity_ = new List<LegalEntity>();
                foreach (XmlNode item in excludedReferenceEntityNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            excludedReferenceEntityIDRef_ = item.Attributes["id"].Value;
                            excludedReferenceEntity_.Add(new LegalEntity(item));
                            IDManager.SetID(excludedReferenceEntityIDRef_, excludedReferenceEntity_[excludedReferenceEntity_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            excludedReferenceEntityIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        excludedReferenceEntity_.Add(new LegalEntity(item));
                        }
                    }
                    else
                    {
                        excludedReferenceEntity_.Add(new LegalEntity(item));
                    }
                }
            }
            
        
            XmlNode trancheNode = xmlNode.SelectSingleNode("tranche");
            
            if (trancheNode != null)
            {
                if (trancheNode.Attributes["href"] != null || trancheNode.Attributes["id"] != null) 
                {
                    if (trancheNode.Attributes["id"] != null) 
                    {
                        trancheIDRef_ = trancheNode.Attributes["id"].Value;
                        Tranche ob = new Tranche(trancheNode);
                        IDManager.SetID(trancheIDRef_, ob);
                    }
                    else if (trancheNode.Attributes["href"] != null)
                    {
                        trancheIDRef_ = trancheNode.Attributes["href"].Value;
                    }
                    else
                    {
                        tranche_ = new Tranche(trancheNode);
                    }
                }
                else
                {
                    tranche_ = new Tranche(trancheNode);
                }
            }
            
        
            XmlNode settledEntityMatrixNode = xmlNode.SelectSingleNode("settledEntityMatrix");
            
            if (settledEntityMatrixNode != null)
            {
                if (settledEntityMatrixNode.Attributes["href"] != null || settledEntityMatrixNode.Attributes["id"] != null) 
                {
                    if (settledEntityMatrixNode.Attributes["id"] != null) 
                    {
                        settledEntityMatrixIDRef_ = settledEntityMatrixNode.Attributes["id"].Value;
                        SettledEntityMatrix ob = new SettledEntityMatrix(settledEntityMatrixNode);
                        IDManager.SetID(settledEntityMatrixIDRef_, ob);
                    }
                    else if (settledEntityMatrixNode.Attributes["href"] != null)
                    {
                        settledEntityMatrixIDRef_ = settledEntityMatrixNode.Attributes["href"].Value;
                    }
                    else
                    {
                        settledEntityMatrix_ = new SettledEntityMatrix(settledEntityMatrixNode);
                    }
                }
                else
                {
                    settledEntityMatrix_ = new SettledEntityMatrix(settledEntityMatrixNode);
                }
            }
            
        
        }
        
    
        #region IndexName_
        private IndexName indexName_;
        public IndexName IndexName_
        {
            get
            {
                if (this.indexName_ != null)
                {
                    return this.indexName_; 
                }
                else if (this.indexNameIDRef_ != null)
                {
                    indexName_ = IDManager.getID(indexNameIDRef_) as IndexName;
                    return this.indexName_; 
                }
                else
                {
                      return this.indexName_; 
                }
            }
            set
            {
                if (this.indexName_ != value)
                {
                    this.indexName_ = value;
                }
            }
        }
        #endregion
        
        public string indexNameIDRef_ { get; set; }
        #region IndexId_
        private List<IndexId> indexId_;
        public List<IndexId> IndexId_
        {
            get
            {
                if (this.indexId_ != null)
                {
                    return this.indexId_; 
                }
                else if (this.indexIdIDRef_ != null)
                {
                    return this.indexId_; 
                }
                else
                {
                      return this.indexId_; 
                }
            }
            set
            {
                if (this.indexId_ != value)
                {
                    this.indexId_ = value;
                }
            }
        }
        #endregion
        
        public string indexIdIDRef_ { get; set; }
        #region IndexSeries_
        private XsdTypePositiveInteger indexSeries_;
        public XsdTypePositiveInteger IndexSeries_
        {
            get
            {
                if (this.indexSeries_ != null)
                {
                    return this.indexSeries_; 
                }
                else if (this.indexSeriesIDRef_ != null)
                {
                    indexSeries_ = IDManager.getID(indexSeriesIDRef_) as XsdTypePositiveInteger;
                    return this.indexSeries_; 
                }
                else
                {
                      return this.indexSeries_; 
                }
            }
            set
            {
                if (this.indexSeries_ != value)
                {
                    this.indexSeries_ = value;
                }
            }
        }
        #endregion
        
        public string indexSeriesIDRef_ { get; set; }
        #region IndexAnnexVersion_
        private XsdTypePositiveInteger indexAnnexVersion_;
        public XsdTypePositiveInteger IndexAnnexVersion_
        {
            get
            {
                if (this.indexAnnexVersion_ != null)
                {
                    return this.indexAnnexVersion_; 
                }
                else if (this.indexAnnexVersionIDRef_ != null)
                {
                    indexAnnexVersion_ = IDManager.getID(indexAnnexVersionIDRef_) as XsdTypePositiveInteger;
                    return this.indexAnnexVersion_; 
                }
                else
                {
                      return this.indexAnnexVersion_; 
                }
            }
            set
            {
                if (this.indexAnnexVersion_ != value)
                {
                    this.indexAnnexVersion_ = value;
                }
            }
        }
        #endregion
        
        public string indexAnnexVersionIDRef_ { get; set; }
        #region IndexAnnexDate_
        private XsdTypeDate indexAnnexDate_;
        public XsdTypeDate IndexAnnexDate_
        {
            get
            {
                if (this.indexAnnexDate_ != null)
                {
                    return this.indexAnnexDate_; 
                }
                else if (this.indexAnnexDateIDRef_ != null)
                {
                    indexAnnexDate_ = IDManager.getID(indexAnnexDateIDRef_) as XsdTypeDate;
                    return this.indexAnnexDate_; 
                }
                else
                {
                      return this.indexAnnexDate_; 
                }
            }
            set
            {
                if (this.indexAnnexDate_ != value)
                {
                    this.indexAnnexDate_ = value;
                }
            }
        }
        #endregion
        
        public string indexAnnexDateIDRef_ { get; set; }
        #region IndexAnnexSource_
        private IndexAnnexSource indexAnnexSource_;
        public IndexAnnexSource IndexAnnexSource_
        {
            get
            {
                if (this.indexAnnexSource_ != null)
                {
                    return this.indexAnnexSource_; 
                }
                else if (this.indexAnnexSourceIDRef_ != null)
                {
                    indexAnnexSource_ = IDManager.getID(indexAnnexSourceIDRef_) as IndexAnnexSource;
                    return this.indexAnnexSource_; 
                }
                else
                {
                      return this.indexAnnexSource_; 
                }
            }
            set
            {
                if (this.indexAnnexSource_ != value)
                {
                    this.indexAnnexSource_ = value;
                }
            }
        }
        #endregion
        
        public string indexAnnexSourceIDRef_ { get; set; }
        #region ExcludedReferenceEntity_
        private List<LegalEntity> excludedReferenceEntity_;
        public List<LegalEntity> ExcludedReferenceEntity_
        {
            get
            {
                if (this.excludedReferenceEntity_ != null)
                {
                    return this.excludedReferenceEntity_; 
                }
                else if (this.excludedReferenceEntityIDRef_ != null)
                {
                    return this.excludedReferenceEntity_; 
                }
                else
                {
                      return this.excludedReferenceEntity_; 
                }
            }
            set
            {
                if (this.excludedReferenceEntity_ != value)
                {
                    this.excludedReferenceEntity_ = value;
                }
            }
        }
        #endregion
        
        public string excludedReferenceEntityIDRef_ { get; set; }
        #region Tranche_
        private Tranche tranche_;
        public Tranche Tranche_
        {
            get
            {
                if (this.tranche_ != null)
                {
                    return this.tranche_; 
                }
                else if (this.trancheIDRef_ != null)
                {
                    tranche_ = IDManager.getID(trancheIDRef_) as Tranche;
                    return this.tranche_; 
                }
                else
                {
                      return this.tranche_; 
                }
            }
            set
            {
                if (this.tranche_ != value)
                {
                    this.tranche_ = value;
                }
            }
        }
        #endregion
        
        public string trancheIDRef_ { get; set; }
        #region SettledEntityMatrix_
        private SettledEntityMatrix settledEntityMatrix_;
        public SettledEntityMatrix SettledEntityMatrix_
        {
            get
            {
                if (this.settledEntityMatrix_ != null)
                {
                    return this.settledEntityMatrix_; 
                }
                else if (this.settledEntityMatrixIDRef_ != null)
                {
                    settledEntityMatrix_ = IDManager.getID(settledEntityMatrixIDRef_) as SettledEntityMatrix;
                    return this.settledEntityMatrix_; 
                }
                else
                {
                      return this.settledEntityMatrix_; 
                }
            }
            set
            {
                if (this.settledEntityMatrix_ != value)
                {
                    this.settledEntityMatrix_ = value;
                }
            }
        }
        #endregion
        
        public string settledEntityMatrixIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

